### PR TITLE
feat: add ASSERT_TRUTH identifier

### PR DESCRIPTION
**Pull request by Reinis-FRP**
Signed-off-by: Reinis Martinsons <reinis@umaproject.org>

### CODE REVIEW

File: UMIPs/umip-assert-truth.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+| UMIP Title     | Add `ASSERT_TRUTH` as a supported price identifier                       |
+| Authors        | Reinis Martinsons (reinis@umaproject.org)                                |
+| Status         | Draft                                                                    |
+| Created        | January 30th, 2023                                                       |
+| Discourse Link | https://discourse.umaproject.org/t/feat-add-assert-truth-identifier/1900 |
+
+# Summary
+
+The DVM should support price requests for the `ASSERT_TRUTH` price identifier. `ASSERT_TRUTH` is intended to be used as
+ a default price identifier for UMA's [Optimistic Asserter contract](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol)
```
Comment (chrismaree) [inline UMIPs/umip-assert-truth.md:15]: it'll be useful to link to the OA UMIP, once it exists, here.

File: UMIPs/umip-assert-truth.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ that allows asserters to make claims about the state of the world. This UMIP does not attempt to put any restrictions
+ on the types of claims that can be made, but rather focuses on the mechanics of how the DVM should be used to resolve
+ disputes over such claims.
+
+Price settlement can happen only in one of two ways:
+
+- Return the `1` value if the claim is true.
```
Comment (chrismaree) [inline UMIPs/umip-assert-truth.md:22]: is it not `1e18`? should we include the scaling here?

Comment (Reinis-FRP) [reply UMIPs/umip-assert-truth.md:22]: Normally price identifiers discuss human readable price values. The scaling is covered below in implementation section.

File: UMIPs/umip-assert-truth.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+# Rationale
+
+This construction sacrifices assurances of determinism in favor of greater price identifier flexibility. This places the
+ burden of correct construction on claims being asserted, but, in return, allows for quicker and easier development
+ without needing to pass through UMA governance for each additional application that builds on top of Optimistic
+ Asserter contract.
+
+# Implementation
+
+1. Voters should identify the Optimistic Asserter contract that was used to make and dispute the assertion. This can be
```
Comment (chrismaree) [inline UMIPs/umip-assert-truth.md:69]: this does assume the OA does not change between when the assertion is made and when a voter tries to find the associated OA address.

File: UMIPs/umip-assert-truth.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ done by calling `getImplementationAddress` on the `Finder` contract with the `OptimisticAsserter` identifier encoded as
+ `bytes32` at the time of assertion. The address of the `Finder` contract depends on the network where the dispute was
+ raised and is available in the `networks` directory of the UMA [protocol repository](https://github.com/UMAprotocol/protocol/tree/master/packages/core/networks).
+2. Voters should decode the ancillary data of the request and determine its `assertionId` parameter.
+3. Voters should search the `AssertionMade` event emitted by the Optimistic Asserter contract from Step 1 for an
+ assertion with the matching `assertionId` parameter from Step 2. Take a note of the timestamp of this event and its
+ `claim` parameter.
```
Comment (chrismaree) [inline UMIPs/umip-assert-truth.md:76]:
```suggestion
 `claim` parameter, as well as any other relevant fields needed to resolve the dispute.
```

File: UMIPs/umip-assert-truth.md (hunk 3 above)
Comment (Reinis-FRP) [reply UMIPs/umip-assert-truth.md:69]: yes, that's why `getImplementationAddress` on the `Finder` should be made at the time of assertion

File: UMIPs/umip-assert-truth.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ is not a string that can be asserted truthful, and voters should return the `0` value.
+5. The decoded `claim` string from Step 4 could include references to the timestamp that was used to make the assertion.
+ If not present, voters should assume that the timestamp of the `AssertionMade` event from Step 3 should be used.
+6. Voters should attempt to evaluate whether the decoded `claim` from Step 4 represents the true state of the world as
+ of claim's timestamp determined from Step 5:
+    - If the claim was true, voters should return the `1` value.
+    - If the claim was false, voters should return the `0` value.
+    - If the claim cannot be unambiguously resolved, voters should return the `0` value.
+
+All resolved price values should be scaled by `1e18` when interacting with contracts directly (e.g. writing scripts,
```
Comment (chrismaree) [inline UMIPs/umip-assert-truth.md:87]: if we just say `1e18` throughout then we could likely ditch this paragraph.

File: UMIPs/umip-assert-truth.md (hunk 2 above)
Comment (chrismaree) [reply UMIPs/umip-assert-truth.md:22]: o ok that' fine.

File: UMIPs/umip-assert-truth.md (hunk 5 above)
Comment (chrismaree) [reply UMIPs/umip-assert-truth.md:87]: can also scratch this given your previous comment.

Comment (Reinis-FRP) [reply UMIPs/umip-assert-truth.md:87]: That would confuse voters if UI abstracts scaling away

**Review by smb2796 — APPROVED** [review]

**Pull request merged**